In [3]:
import pandas as pd

In [4]:
claims= pd.read_csv('/Users/blessingmashonga/Downloads/healthcare-claims-where-is-the-money-going-tvhlq/claims.csv')

In [5]:
members= pd.read_csv('/Users/blessingmashonga/Downloads/healthcare-claims-where-is-the-money-going-tvhlq/members.csv')

In [6]:
claims.shape

(449, 9)

In [7]:
members.shape

(100, 6)

In [9]:
claims.info()

<class 'pandas.DataFrame'>
RangeIndex: 449 entries, 0 to 448
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   claim_id       449 non-null    int64  
 1   member_id      449 non-null    int64  
 2   provider_id    449 non-null    str    
 3   claim_date     449 non-null    str    
 4   claim_type     449 non-null    str    
 5   cpt_code       449 non-null    str    
 6   icd_code       449 non-null    str    
 7   billed_amount  449 non-null    float64
 8   paid_amount    449 non-null    float64
dtypes: float64(2), int64(2), str(5)
memory usage: 31.7 KB


In [10]:
claims.head()

,claim_id,member_id,provider_id,claim_date,claim_type,cpt_code,icd_code,billed_amount,paid_amount
0,1,1,PRV00001,03-12-2023,Outpatient,12345,A12.3,1800.0,1600.0
1,2,1,PRV00002,07-22-2023,Inpatient,67890,B99.4,32000.0,28000.0
2,3,1,PRV00003,01-15-2024,Lab,23456,C50.1,450.0,400.0
3,4,1,PRV00004,09-05-2023,Pharmacy,34567,D12.7,120.0,120.0
4,5,2,PRV00123,04-12-2023,Outpatient,12345,A12.3,2100.0,1680.0


In [11]:
members.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype
---  ------                 --------------  -----
 0   member_id              100 non-null    int64
 1   member_age             100 non-null    int64
 2   member_gender          100 non-null    str  
 3   plan_type              100 non-null    str  
 4   enrollment_start_date  100 non-null    str  
 5   enrollment_end_date    85 non-null     str  
dtypes: int64(2), str(4)
memory usage: 4.8 KB


In [12]:
members.head()

,member_id,member_age,member_gender,plan_type,enrollment_start_date,enrollment_end_date
0,1,45,F,PPO,5/14/2021,11/30/2023
1,2,52,M,HMO,3/22/2020,8/15/2022
2,3,38,F,EPO,1/10/2022,4/1/2024
3,4,61,M,POS,9/5/2020,12/1/2022
4,5,29,F,PPO,2/28/2023,NaN


In [16]:
claims.duplicated().sum()

np.int64(0)

In [17]:
members.duplicated().sum()

np.int64(0)

In [18]:
members['member_id'].nunique()

100

In [19]:
members['plan_type'].unique()

<StringArray>
['PPO', 'HMO', 'EPO', 'POS']
Length: 4, dtype: str

In [20]:
members['member_gender'].unique()

<StringArray>
['F', 'M']
Length: 2, dtype: str

In [37]:
members['enrollment_end_date'].isnull().sum()

np.int64(15)

In [26]:
members[members['enrollment_end_date'].isnull()]

,member_id,member_age,member_gender,plan_type,enrollment_start_date,enrollment_end_date
4,5,29,F,PPO,2/28/2023,NaN
11,12,52,M,HMO,2/1/2021,NaN
15,16,41,M,HMO,1/12/2023,NaN
19,20,35,M,EPO,12/1/2020,NaN
25,26,40,M,HMO,4/12/2023,NaN
39,40,41,M,PPO,6/19/2022,NaN
45,46,70,M,PPO,2/14/2023,NaN
49,50,47,M,EPO,5/3/2023,NaN
55,56,21,M,HMO,1/16/2023,NaN
65,66,72,M,HMO,2/1/2023,NaN


In [27]:
members['enrollment_start_date']=pd.to_datetime(members['enrollment_start_date'])

In [28]:
members['enrollment_end_date']=pd.to_datetime(members['enrollment_end_date'])

In [29]:
claims['claim_date']=pd.to_datetime(claims['claim_date'])

In [30]:
claims.info()

<class 'pandas.DataFrame'>
RangeIndex: 449 entries, 0 to 448
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   claim_id       449 non-null    int64         
 1   member_id      449 non-null    int64         
 2   provider_id    449 non-null    str           
 3   claim_date     449 non-null    datetime64[us]
 4   claim_type     449 non-null    str           
 5   cpt_code       449 non-null    str           
 6   icd_code       449 non-null    str           
 7   billed_amount  449 non-null    float64       
 8   paid_amount    449 non-null    float64       
dtypes: datetime64[us](1), float64(2), int64(2), str(4)
memory usage: 31.7 KB


In [32]:
members.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   member_id              100 non-null    int64         
 1   member_age             100 non-null    int64         
 2   member_gender          100 non-null    str           
 3   plan_type              100 non-null    str           
 4   enrollment_start_date  100 non-null    datetime64[us]
 5   enrollment_end_date    85 non-null     datetime64[us]
dtypes: datetime64[us](2), int64(2), str(2)
memory usage: 4.8 KB


In [41]:
claims

,claim_id,member_id,provider_id,claim_date,claim_type,cpt_code,icd_code,billed_amount,paid_amount
0,1,1,PRV00001,2023-03-12,Outpatient,12345,A12.3,1800.0,1600.0
1,2,1,PRV00002,2023-07-22,Inpatient,67890,B99.4,32000.0,28000.0
2,3,1,PRV00003,2024-01-15,Lab,23456,C50.1,450.0,400.0
3,4,1,PRV00004,2023-09-05,Pharmacy,34567,D12.7,120.0,120.0
4,5,2,PRV00123,2023-04-12,Outpatient,12345,A12.3,2100.0,1680.0
...,...,...,...,...,...,...,...,...,...
444,460,100,PRV01234,2023-02-15,Outpatient,12345,A12.3,1800.0,1500.0
445,461,100,PRV04789,2023-03-22,Inpatient,54321,B99.4,21000.0,14000.0
446,462,100,PRV02345,2023-07-11,Lab,67890,C21.1,320.0,320.0
447,463,100,PRV09876,2023-11-08,Pharmacy,11111,D45.6,120.0,100.0


In [64]:
claims[['billed_amount','paid_amount']].describe()

,billed_amount,paid_amount
count,449.000000,449.000000
mean,4592.795212,3453.774855
std,6747.042564,5040.669863
min,60.000000,0.000000
25%,260.000000,210.000000
50%,1800.000000,1280.000000
75%,5400.000000,4200.000000
max,42000.000000,35000.000000


In [80]:
claims[(~claims['member_id'].isin(members['member_id']))]

,claim_id,member_id,provider_id,claim_date,claim_type,cpt_code,icd_code,billed_amount,paid_amount


In [84]:
members[~members['member_id'].isin(claims['member_id'])]

,member_id,member_age,member_gender,plan_type,enrollment_start_date,enrollment_end_date


In [85]:
claims['claim_type'].value_counts()

claim_type
Outpatient    105
Inpatient      99
Emergency      88
Pharmacy       81
Lab            76
Name: count, dtype: int64

In [10]:
paid_amount_by_claim_type=claims.groupby('claim_type')['paid_amount'].sum()
paid_amount_by_claim_type

claim_type
Emergency      294441.36
Inpatient     1092456.00
Lab             23412.35
Outpatient     129052.75
Pharmacy        11382.45
Name: paid_amount, dtype: float64

In [13]:
claims.groupby('claim_type')['billed_amount'].sum()

claim_type
Emergency      384241.55
Inpatient     1478601.25
Lab             25789.90
Outpatient     160717.75
Pharmacy        12814.60
Name: billed_amount, dtype: float64

In [14]:
claims.groupby('claim_type')['claim_id'].count()

claim_type
Emergency      88
Inpatient      99
Lab            76
Outpatient    105
Pharmacy       81
Name: claim_id, dtype: int64

In [22]:
claim_type_cost_breakdown= claims .groupby('claim_type').agg({'billed_amount':'sum','claim_id':'count','paid_amount':'sum'}).sort_values('paid_amount', ascending=False)

,billed_amount,claim_id,paid_amount
claim_type,,,
Inpatient,1478601.25,99,11034.909091
Emergency,384241.55,88,3345.924545
Outpatient,160717.75,105,1229.073810
Lab,25789.90,76,308.057237
Pharmacy,12814.60,81,140.524074


In [23]:
claims.groupby('claim_type')['paid_amount'].mean()

claim_type
Emergency      3345.924545
Inpatient     11034.909091
Lab             308.057237
Outpatient     1229.073810
Pharmacy        140.524074
Name: paid_amount, dtype: float64

## Q1 Findings

* **Inpatient claims are the largest driver of insurer-paid costs**, with total paid costs of **$1.09 million**.
* Inpatient claims accounted for **99 claims**, compared with 105 outpatient claims and 88 emergency claims.
* Despite not having the highest claim volume, inpatient claims had the **highest average paid amount per claim at $11,034.91**.
* **Emergency claims** were the second-largest contributor to total paid costs, with **$294,441.36** paid across 88 claims.
* **Outpatient, Lab, and Pharmacy claims** contributed substantially less to total paid costs.
* Overall, the analysis suggests that **inpatient care is the primary area of concern because of its high cost per claim**, rather than simply having the highest number of claims.



## Question 2 — CPT & ICD Cost Drivers

**Business question:**
Which medical procedures and diagnoses are driving the insurer's healthcare costs?

**Objective:**
Identify the top 10 CPT (Current Procedural Terminology) codes and top 10 ICD (International Classification of Diseases) codes by total paid amount. Also identify CPT codes with a high average paid amount per claim.

**Business purpose:**
Determine which specific procedures and diagnoses contribute most to healthcare spending and distinguish between cost drivers caused by high claim volume and those caused by high cost per claim.


In [33]:
top_10_cpt_paid =claims.groupby('cpt_code')['paid_amount'].sum().sort_values(ascending = False).head(10)
top_10_cpt_paid                                                                                                    


cpt_code
67890    242735.00
23456    203790.75
123      122010.00
12345    115236.90
99223     57350.00
34567     54445.00
45678     52054.20
567       38969.00
54321     36600.00
56789     33207.00
Name: paid_amount, dtype: float64

In [35]:
claims.groupby('icd_code')['paid_amount'].sum().sort_values(ascending = False).head(10)

icd_code
I10        259566.00
A12.3      152147.00
B20        140990.00
B20.1      105210.00
C34.91      62905.00
B99.4       51000.00
E11.65      50512.00
J45.909     44465.56
E11.9       34766.00
A01.1       34260.00
Name: paid_amount, dtype: float64

In [36]:
 claims.groupby('cpt_code')['paid_amount'].mean().sort_values(ascending = False).head(10)
                                                                                                    


cpt_code
36512    25600.0
10101    17250.6
89012    15250.0
512      14000.0
50255    13000.0
20610    12000.0
61234    10500.0
99217    10200.0
43752    10200.0
123      10167.5
Name: paid_amount, dtype: float64

## Q2 Findings

* The analysis identified the procedures and diagnoses associated with the highest insurer-paid costs.
* **CPT 67890** was the largest individual procedure cost driver, with **$242,735** in total paid costs, followed by CPT 23456 at **$203,790.75**.
* **ICD I10** was the largest diagnosis-level cost driver, with **$259,566** in total paid costs.
* The highest average paid amount per CPT claim was associated with **CPT 36512**, at **$25,600 per claim**.
* Some CPT codes with high average costs per claim did not appear among the top 10 by total paid amount. This indicates that **high cost per claim and high overall spending are not necessarily driven by the same procedures**.
* **CPT 123** appeared in both the top 10 by total paid amount and the top 10 by average paid amount per claim, making it a notable procedure for further investigation.
* Overall, the analysis shows that healthcare spending is concentrated among a relatively small number of procedures and diagnoses, providing potential areas for the insurer to investig


In [39]:
top_10_members = claims.groupby('member_id')['paid_amount'].sum().sort_values(ascending = False).head(10)
top_10_members 

member_id
6     43300.00
32    40080.00
58    35920.00
82    30690.00
28    30560.00
20    30375.00
71    30230.00
1     30120.00
36    25530.75
8     23046.80
Name: paid_amount, dtype: float64

In [45]:
top_10_members = claims.groupby('member_id')['paid_amount'].sum().sort_values(ascending = False).head(10)
top_member_claims =claims[claims['member_id'].isin(top_10_members.index)]
top_member_claims 

In [49]:
claim_type_by_member=top_member_claims.groupby(['member_id','claim_type'])['paid_amount'].sum()
claim_type_by_member

member_id  claim_type
1          Inpatient     28000.00
           Lab             400.00
           Outpatient     1600.00
           Pharmacy        120.00
6          Emergency      8000.00
           Inpatient     31500.00
           Lab            1200.00
           Outpatient     2600.00
8          Emergency      4100.00
           Inpatient     17250.60
           Lab             280.00
           Outpatient     1340.60
           Pharmacy         75.60
20         Emergency      3000.00
           Inpatient     25050.00
           Lab             465.00
           Outpatient     1710.00
           Pharmacy        150.00
28         Emergency       960.00
           Inpatient     25600.00
           Lab             280.00
           Outpatient     3600.00
           Pharmacy        120.00
32         Emergency      3200.00
           Inpatient     35000.00
           Lab             320.00
           Outpatient     1500.00
           Pharmacy         60.00
36         Emergency      

In [64]:
claim_verification = claim_type_by_member.reset_index()
claim_verification[claim_verification['member_id'] == 6]['paid_amount'].sum()

np.float64(43300.0)

## Q3 Findings

* **Member 6 is the highest-cost member**, with total insurer-paid claims of **$43,300**.
* **Inpatient claims are the primary cost driver** across all 10 highest-cost members.
* The concentration of spending in inpatient claims suggests that **inpatient utilization and costs should be a priority area for further investigation**.
* The insurer should investigate whether high inpatient spending is driven by **specific procedures, diagnoses, providers, or unusually expensive claims**.


In [66]:
claims['paid_ratio']=claims['paid_amount']/claims['billed_amount']

In [67]:
claims['paid_ratio'].describe()

count    449.000000
mean       0.831135
std        0.139702
min        0.000000
25%        0.768293
50%        0.800000
75%        1.000000
max        1.000000
Name: paid_ratio, dtype: float64

In [69]:
overall_paid_ratio= claims['paid_amount'].sum()/claims['billed_amount'].sum()

In [80]:
claim_type_expense=claims.groupby('claim_type')[['paid_amount','billed_amount']].sum().reset_index()
claim_type_expense['paid_ratio'] = claim_type_expense['paid_amount']/claim_type_expense['billed_amount']
claim_type_expense.sort_values('paid_ratio', ascending = False)

,claim_type,paid_amount,billed_amount,paid_ratio
2,Lab,23412.35,25789.90,0.907811
4,Pharmacy,11382.45,12814.60,0.888241
3,Outpatient,129052.75,160717.75,0.802978
0,Emergency,294441.36,384241.55,0.766292
1,Inpatient,1092456.00,1478601.25,0.738844


In [89]:
claim_type_expense=claims.groupby('claim_type').agg({'paid_amount':'sum','billed_amount':'sum','claim_id':'count'}).reset_index()
claim_type_expense['paid_ratio'] = claim_type_expense['paid_amount']/claim_type_expense['billed_amount']
claim_type_expense['unpaid_amount']= claim_type_expense['billed_amount']-claim_type_expense['paid_amount']
claim_type_expense.sort_values('paid_ratio', ascending = False)

,claim_type,paid_amount,billed_amount,claim_id,paid_ratio,unpaid_amount
2,Lab,23412.35,25789.90,76,0.907811,2377.55
4,Pharmacy,11382.45,12814.60,81,0.888241,1432.15
3,Outpatient,129052.75,160717.75,105,0.802978,31665.00
0,Emergency,294441.36,384241.55,88,0.766292,89800.19
1,Inpatient,1092456.00,1478601.25,99,0.738844,386145.25
